.github/workflows/auto-pr.yaml

```diff
@@ -0,0 +1,97 @@
+        with:
+          # This is necessary to avoid unexpected auto-merge in git cherry-pick
+          fetch-depth: 0
+
+      - name: Create pull requests
+        run: |
+          # Get the author of the merged PR
+          assignee=$(gh api graphql -F owner=${{ github.event.repository.owner.login }} -F repoName=${{ github.event.repository.name }} -F userName=${{ github.event.pull_request.user }} -f query='
+          query($owner: String!, $repoName: String!, $userName: String!) {
+            repository(owner: $owner, name: $repoName) {
+              assignableUsers(first: 100, query: $userName) {
+                nodes {
+                  name,
+                  email
+                }
+              }
+            }
+          }' | jq -r '.data.repository.assignableUsers.nodes[0]'
+          )
+
+          # Get GitHub projects associated with the merged PR and extract full versions (e.g. "3.7.1")
+          versions=$(gh api graphql -F owner=${{ github.event.repository.owner.login }} -F repoName=${{ github.event.repository.name }} -F pullRequestId=${{ github.event.number }} -f query='
+          query($owner: String!, $repoName: String!, $pullRequestId: Int!) {
+            repository(owner: $owner, name: $repoName) {
+              pullRequest(number: $pullRequestId) {
+                projectsV2(first: 100) {
+                  nodes {
+                    title
+                  }
+                }
+              }
+            }
+          }
+          ' | jq -r '.data.repository.pullRequest.projectsV2.nodes[].title' | sed -n 's/^ScalarDB \([0-9]*\.[0-9]*\.[0-9]*\).*/\1/p'
+          )
+          echo -------------
+          echo "versions: $versions"
+          echo -------------
+
+          # Extract target support branch and release branch names from the full versions
+          branches=$(echo $versions | ruby -e '
+            require "set"
+
+            versions = STDIN.read.strip.split
+
+            versions.inject(Set.new) {|acc, v|
+              # Drop patch version by getting the first 2 elements
+              parts = v.strip.split(".")[0, 2]
+              # Support branch
+              acc << parts[0]
+              # Release branch
+              acc << parts.join(".")
+              acc
+            }.each {|branch|
+              puts branch
+            }
+          ')
+          echo -------------
+          echo "branches: $branches"
+          echo -------------
+
+          # Create PRs based on the target support branches and release branches
+          git config --global user.email $(echo $assignee | jq -r .email)
+          git config --global user.name $(echo $assignee | jq -r .name)
+          git fetch origin
+          for branch in $branches; do
+            git checkout $branch
+            new_branch="$branch-pull-${{ github.event.number }}"
+            git checkout -b "$new_branch"
+            git status
+            git cherry-pick --no-rerere-autoupdate -m1 ${{ github.sha }}
```

Comment:
We just manually create all the PRs if this `cherry-pick` fails?
Or, can we skip failed `cherry-pick` and create PRs for whatever succeeded? (and manually create PRs for failed ones?)

Reply:
Good point. The current workflow stops creating PRs once cherry-pick fails. I first thought just skipping a failure and continuing following branches might let ScalarDB developers miss the skipped branch. So I avoided skipping a failure. But I just noticed even the current logic would still cause partially succeeded and failed branches... That's not good.

I think a reasonable option is to create a GitHub issue assigning the original PR author for a failed PR creation and continue next branches. I'll try this.

Reply:
Addressed it in https://github.com/scalar-labs/scalardb/pull/884/commits/ce4e2b583d4319b3ed3ed4b81953b57fa5ce839d